Sketching virtual network with loader shimming

ef4 opened the pull request.

ef4 committed:
Sketching virtual network with loader shimming

jurgenwerk committed:
Wire up virtual network with the loaders (without shimming handlers for now)

jurgenwerk committed:
Prototyping: shimmed externals via virtual network

lukemelia committed:
Get shimmed externals via virtual network working

jurgenwerk committed:
Fix host tests

jurgenwerk committed:
Fix tests after shim externals refactor

jurgenwerk committed:
Merge branch 'main' into virtual-network

lukemelia commented on packages/host/tests/integration/search-index-test.gts:
```diff
@@ -2668,8 +2653,23 @@ module('Integration | search-index', function (hooks) {
+        'https://shimmed-module/ember-concurrency/-private/async-arrow-runtime',
+        'https://shimmed-module/ember-modifier',
+        'https://shimmed-module/lodash',
+        'https://shimmed-module/tracked-built-ins',
```
I'm not sure if we want this

jurgenwerk committed:
Implement loader-specific module shimming via virtual network

jurgenwerk committed:
Merge branch 'main' into virtual-network

ef4 committed:
move shared module-scoped state to globals

This is a followup to the change that switch to Symbol.for, with the intent of making it not matter which loader you retrive cardApi from. It turns out we also need to make sure you're sharing the same module-scoped WeakMaps, etc.

jurgenwerk committed:
Fixes all tests but one

jurgenwerk committed:
Also add card tracking to globals

jurgenwerk committed:
Fix the rest of the tests

jurgenwerk committed:
Merge branch 'main' into virtual-network

jurgenwerk committed:
Merge branch 'main' into virtual-network

jurgenwerk committed:
Fix another test, one to go

jurgenwerk committed:
Fix a situation where creating a linked card is not reflected in the parent card

jurgenwerk committed:
Add a note about possible race condition

jurgenwerk committed:
Move packages fake origin to virtual network

jurgenwerk committed:
Not needed
